ml-agents/mlagents/trainers/trainer.py

```diff
@@ -282,3 +282,8 @@ def update_policy(self):
         Uses demonstration_buffer to update model.
         """
         raise UnityTrainerException("The update_model method was not implemented.")
+
+    def clear_update_buffer(self) -> None:
+        raise UnityTrainerException(
```

Comment:
@ervteng Not sure what we should do about this. This line (correctly) complains
https://github.com/Unity-Technologies/ml-agents/blob/40d74dc1125f69b0af7e14f216a6dcea87e20b5c/ml-agents/mlagents/trainers/trainer_controller.py#L298
because clear_update_buffer() was only on RLTrainer, not Trainer. I figure it might be a moot point after the BC removal - is part of that PR combining Trainer and RLTrainer?

Reply:
The BC remove doesn't combine the two - I think the right thing is to put it only in rl_trainer, since there's no "update buffer" per se for all trainers

Reply:
The what should the TrainerController call? It only knows about Trainers, not RLTrainers.